src/ansys/sound/core/spectral_processing/filter.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Filter a signal."""
+
+import warnings
+
+from ansys.dpf.core import Field, Operator
+import matplotlib.pyplot as plt
+import numpy as np
+
+from . import SpectralProcessingParent
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+
+ID_OPERATOR_DESIGN = "filter_design_minimum_phase_FIR_filter_from_FRF"
+ID_OPERATOR_LOAD = "load_FRF_from_txt"
+ID_OPERATOR_FILTER = "filter_signal"
+
+
+class Filter(SpectralProcessingParent):
+    """Filter class.
+
+    This class allows designing, loading, and applying a digital filter to a signal. The filter
+    coefficients can be provided directly, or computed from a specific frequency response function
+    (FRF). In this latter case, the filter is designed as a minimum-phase FIR filter, and the
+    filter denominator is set to 1 as a consequence.
+
+    .. note::
+
+        Whether they are derived from the provided FRF or specified directly, the filter
+        coefficients are linked to the sampling frequency value that is given in the argument
+        ``sampling_frequency`` of the ``Filter`` class. As a consequence, the signal to filter
+        ``signal`` must have the same sampling frequency. If necessary, use the :class:`.Resample`
+        class to resample the signal prior to using the ``Filter`` class.
+    """
+
+    def __init__(
+        self,
+        a_coefficients: list[float] = [],
```

Comment:
@ansaminard I would preferably put b_coefficients BEFORE a_coeffcients in the ctor signature, as it is usually done (ex: Matlab, SciPy)

Reply:
done